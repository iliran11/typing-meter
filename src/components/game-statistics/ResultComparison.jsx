import React from 'react';
import Tooltip from '../Tooltip';
import gameStats from './gameStats';
import { shape,number } from 'prop-types';

class ResultComparison extends React.Component {
  constructor(props) {
    super(props);
    this.width = 100;
  }
  componentDidMount() {
    this.width = 100 - this.percentile;
  }
  get graphStyle() {
    return {
      width: `${this.width}%`
    };
  }
  get percentile() {
    return gameStats(this.props.result).percentile;
  }
  render() {
    return (
      <div className="result-comparison">
        <div className="expanding-graph">
          <div className="revealer" style={this.graphStyle}>
            <Tooltip result={this.percentile} />
          </div>
        </div>
      </div>
    );
  }
}

export default ResultComparison;
