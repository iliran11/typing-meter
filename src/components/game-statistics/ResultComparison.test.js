import React from 'react';
import ResultComparison from './ResultComparison';
import { shallow } from 'enzyme';

const props = {
  result: {
    percentile: 10
  }
};
describe('`ResultComparison`', () => {
  it('renders properly', () => {
    const wrapper = shallow(<ResultComparison {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
