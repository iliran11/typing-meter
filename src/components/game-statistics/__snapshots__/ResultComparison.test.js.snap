// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`ResultComparison\` renders properly 1`] = `
<div
  className="result-comparison"
>
  <div
    className="expanding-graph"
  >
    <div
      className="revealer"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Tooltip
        result={98}
      />
    </div>
  </div>
  <div
    className="small-label"
  >
    Where You Stand Against Others
  </div>
</div>
`;
